### ISSUE TITLE
Can changes in annotation be detected?

### ISSUE THREAD

**Issue by debraj-manna** [lvc/japi-compliance-checker#46]
Can `japi-compliance-checker` be used to detect changes (both addition or removal) in annotation in a java class?

**Comment by lvc** [comment]
Dangerous changes in annotations are detected currently:

1. Removed annotations
2. Added/changed/removed annotation elements

See complete list of _\*Annotation\*_  related rules in the `modules/RulesSrc.xml` and `modules/RulesBin.xml`.

Added annotations are not reported yet (this may go to 'Other changes in Data Types' section of the report in future versions of the tool).

Thanks.

**Comment by debraj-manna** [comment]
To test the annotation I created a simple jar with just one class like below

```
package org.debraj.pkgcmp.dummy;

import com.google.common.eventbus.Subscribe;

public class PackageAnnotation {
	@Subscribe
	public void dummyMethod() { }
}
```
I created one jar by adding `@Subscribe` and then another jar by removing annotation. But tool was not able to detect changes.

```
manna-m01:japi-compliance-checker dmanna$ ./japi-compliance-checker.pl ~/Desktop/pkgcmp-0.0.1-SNAPSHOT.jar ~/Desktop/pkgcmp-0.0.2-SNAPSHOT.jar
Preparing, please wait ...
Using Java 1.8.0_131
Reading classes 0.0.1-SNAPSHOT ...
Reading classes 0.0.2-SNAPSHOT ...
Comparing classes ...
Creating compatibility report ...
Binary compatibility: 100%
Source compatibility: 100%
Total binary compatibility problems: 0, warnings: 0
Total source compatibility problems: 0, warnings: 0
Report: compat_reports/pkgcmp/0.0.1-SNAPSHOT_to_0.0.2-SNAPSHOT/compat_report.html
```
 

The pom.xml is like below
```
<project xmlns="http://maven.apache.org/POM/4.0.0" xmlns:xsi="http://www.w3.org/2001/XMLSchema-instance"
  xsi:schemaLocation="http://maven.apache.org/POM/4.0.0 http://maven.apache.org/xsd/maven-4.0.0.xsd">
  <modelVersion>4.0.0</modelVersion>

  <groupId>org.debraj</groupId>
  <artifactId>pkgcmp</artifactId>
  <version>0.0.1-SNAPSHOT</version>
  <packaging>jar</packaging>

  <name>pkgcmp</name>
  <url>http://maven.apache.org</url>

  <properties>
    <project.build.sourceEncoding>UTF-8</project.build.sourceEncoding>
  </properties>

  <dependencies>
    <dependency>
    <groupId>com.google.guava</groupId>
    <artifactId>guava</artifactId>
    <version>23.5-jre</version>
</dependency>
  </dependencies>
</project>
```

**Comment by debraj-manna** [comment]
@lvc - Any thoughts on this?

**Comment by lvc** [comment]
@debraj-manna 

The tool detects removed annotation definitions, not the removed use of the annotation. Applying of annotations doesn't affect source and binary compatibility. It may affect behavioral compatibility that is out of the scope of the tool analysis.

https://wiki.eclipse.org/Evolving_Java-based_APIs_2#Evolving_annotations_on_API_elements

Thanks.
